impl/integration/main_test.go

```diff
@@ -48,25 +55,42 @@ func TestIntegration(t *testing.T) {
 				// canceling the master context.
 				ctx, cancel := context.WithCancel(ctx)
 				defer cancel()
-				test.Fn(ctx, env.Env, t)
+				resps := test.Fn(ctx, env.Env, t)
+				if *generate && resps != nil {
+					err = SaveTestVectors(*testdataDir, test.DirectoryFilename, test.RespFilename, env.Env, resps)
+					if err != nil {
+						t.Fatalf("saveTestVectors() failed: %v", err)
+					}
+				}
 			}()
 		})
 	}
 }
 
-func TestGenerateVectors(t *testing.T) {
-	flag.Parse()
-	if !*generate {
-		t.Skip("Test vector generation not requested")
+// SaveTestVectors generates test vectors for interoprability testing.
+func SaveTestVectors(dir string, directoryFilename string, responseFilename string, env *integration.Env, resps []testdata.ResponseVector) error {
```

Comment:
When multiple parameters have the same type, the type can be omitted:
```suggestion
func SaveTestVectors(dir, directoryFilename, responseFilename string, env *integration.Env, resps []testdata.ResponseVector) error {
```